auth/providers/azure/azure.go

```diff
@@ -139,8 +132,32 @@ func getMetadata(aadEndpoint, tenantID string) (*metadataJSON, error) {
 		Backoff:      retryablehttp.DefaultBackoff,
 		Logger:       log.Default(),
 	}
+}
 
-	response, err := retryClient.Get(metadataURL)
+// withRetryableHttpClient sets the oauth2.HTTPClient key of the context to an
+// *http.Client made from makeRetryableHttpClient.
+// Some of the libraries we use will take the client out of the context via
+// oauth2.HTTPClient and use it, so this way we can add retries to external code.
+func withRetryableHttpClient(ctx context.Context) context.Context {
+	retryClient := makeRetryableHttpClient()
+	return context.WithValue(ctx, oauth2.HTTPClient, retryClient.StandardClient())
```

Comment:
I see this being used only in vendor/github.com/coreos/go-oidc/oidc.go. Are there other places where client is being read from the context to use the retries?

Reply:
That's the right place, though it's called in a few different ways (`oidc.NewProvider` and by `remoteKeySet.updateKeys`, which happens when we call `s.verifier.Verify(ctx, token)` in `Check`).